### ISSUE TITLE
Cannot get the level to go completey black (dark) in the scene

### ISSUE THREAD

**Issue by casigus** [TheGameCreators/GameGuruRepo#4653]
changing the ambient color does nothing.

no I dont know the steps how to get to this, its just not working and hasn't done for a while for some maps.

![image](https://github.com/TheGameCreators/GameGuruRepo/assets/68471279/4a3c7362-d8ab-4516-b804-6b1b4df4bf81)

**Comment by casigus** [comment]
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/68471279/0c071382-4382-4a49-b449-44d9004a6c17)

for example, set the sun light to 0 and its not changing any of the lightning. Doesn't happen on all maps, but happens on some of them and I can't find a fix

**Comment by davetgc** [comment]
I've tested this, and so has a third party and it seems to be working as expected, can anyone else shed some light on this? Are you also experiencing an issue with the lighting?

**Comment by casigus** [comment]
yeah! for some reason its just not letting me set the lighting to zero ! this map should be completely dark but nope!! its beeen an issue for a while now !!! the lighting works when ramped up to full (so when set the sun to max, it lights up fine, but the second seet it to be super dark, nothing happens) its very odd!

**Comment by synchromesh62** [comment]
You need to turn the white lighting colour down also in the second pic.
It still wont be perfect but im currently testing a way to achieve total darkness.

**Comment by casigus** [comment]
I used to have total darkness, always had it, then boom it just stopped working.


On 24 May 2023, at 01:16, synchromesh62 ***@***.***> wrote:


You need to turn the white lighting colour down also in the second pic.
It still wont be perfect but im currently testing a way to achieve total darkness.

**Comment by synchromesh62** [comment]
You can get total darkness by lowering the exposure setting but to low and you lose flashlight and emissions.

There is actually a very simple solution i came up with.
Add your own **World Probe.**

 The video will firstly show **Default Darkeness** ( which is pretty rubbish ) and the Ambience never truly goes dark.
So i added my own World Probe  adjusting all the settings to 5000000.
It rectifies the Ambience without the need to alter any exposure ..

https://youtu.be/5mbeXVta4is

**Comment by casigus** [comment]
YOUR missing the point. I CANT GET TOTAL DARKNESS. AT ALL.

THIS IS what happens.

https://www.youtube.com/watch?v=w-zooIlRD4E

why I opened the bug report.

**Comment by casigus** [comment]
AYE added a probe! and yeah nothinggggggggggggg. had this bug for a while now :(

**Comment by casigus** [comment]
the ground SHOULD not be glowing 

![image](https://github.com/TheGameCreators/GameGuruRepo/assets/68471279/3a527d47-77c4-4951-9e7f-c133028cc21c)


it should be completely dark. but nope it glows because why wouldn't it?! 

its very annoying.

**Comment by casigus** [comment]
more info: if I create a NEW MAP, i can change the lighting which managed to do as showed here:

![image](https://github.com/TheGameCreators/GameGuruRepo/assets/68471279/8b54b9c0-70dc-48fb-883c-a5818a8c7700)


However. If I open the bugged Map where the lighting doesn't work, close the bugged map then open the new map I created, the lighting bug then occurs in that map too....and it never goes away regardless of if I save it or not....


if I then open or create a new map after having the bugged map open....the bug then carries into that map.

in short. its a broken mess. 

yay.

**Comment by casigus** [comment]
ive tracked it down to how the bug starts, when you generate a map in the terrain editor and it changes the landscape/sky it breaks the lighting settings messing up the ambient colour not doing in anything, which then carries on to other maps if you swap between those maps.

If you save your work and map then the bug saves with it.

**Comment by TenebreGaming** [comment]
I've been having the same issue, and the probe thing worked on a single level (and I was happy thinking it was a soltion) but fails to do anything on a dozen others. Everything is black except the terrain. IMO, this should be considered a show stopper, as many people will want to make dark games that are in an outside environment.

![Screenshot 2023-05-26 192445](https://github.com/TheGameCreators/GameGuruRepo/assets/133550706/b5353ea6-8d91-4eb0-88f1-713b56446026)

**Comment by TenebreGaming** [comment]
I have been able to correct this issue in my time-of-day script by setting an initial value that is dark, then waiting 500ms before initializing it to the actual time I want. If I try doing it faster than that, it doesn't work. If you don't do this, then whatever the initial value is for the ground remains (so, for instance, if there is a glow caused by sun reflection, then it will stay even after setting everything to levels that should be dark.

**Comment by casigus** [comment]
glad you managed to correct it! and yeah i feel its a show stopper for sure.

**Comment by synchromesh62** [comment]
> 

Strange it works every time for me ?
The only thing i make sure i do is set it all as dark as possible ( especially the black sky ) first then add the probe, 
Otherwise the probe reflects and remembers the daytime sky and will not go completely dark.
Once the probe is set in the dark the ambience slider will then go to daylight and dark just fine moving the slider up and down.

**Comment by TenebreGaming** [comment]
I also just noticed (after loading up Max again and going into my level) that it wasn't getting dark again. After reading the comments above, I decided to check a few things. I found that I actually need to be near the probe (in the editor or in game) for the probe to work. If I set the ambient color to black and approach the probe, everything slowly fades to black. This particular map is a 5kx5k map, and the probe is basically in the center, and it's values are all set to 50,000,000, which is pretty extreme and should very well cover the entire map. I can supply video if needed.

The thing is, we shouldn't have to do any of this in order to get a dark map, and it should be fairly simple to fix in the actual code.  I have worked as a game tester and a game programmer for a AAA company before, and I've written enough DirectX and OpenGL code and shaders to be able to say this as fact. The natural state for everything is black, so when the lighting color is black, the ambient color is black, and the sun intensity is 0, whether set in the GUI or via scripting, everything not specifically lit should be dark.

**Comment by Kitakazi** [comment]
Hello wanted to report that I am experiencing this bug, or one similar. 

I have all my lightning set to 0 but there is a strong ambient light coming from somewhere. Completely ruins my map. It's like everything is covered in some soft blue light. This is started a few updates ago but it did NOT use to be like this.

**Comment by Kitakazi** [comment]
Ok weird I just opened "zombie cellar demo - level1" was messing with the lighting in there, trying to reproduce this but I could not reproduce the bug. So I went back to the level from my screenshot above, and the lightning is now back to normal and seems fixed. No idea how that fixed it.

**Comment by Kitakazi** [comment]
Ok something is definitely bugged with the lighting. I've noticed that sometimes opening an FPM will produce this bug while other times opening the same FPM does not produce it.

https://youtu.be/fHyV42PEG5w
 Here in this video I go straight to the Zombie Cellar Demo.FPM and you can see the bug is happening, the level is super bright and that bright blue glow is on everything. 

Then I close Max and reopen the exact same FPM and the bug is gone. Everything is nice and black as it should be.

**Comment by synchromesh62** [comment]
Ye i get this a lot as well. 
Usually restarting Max as you did rectifies it. Very annoying for sure.

**Comment by LeeBamberTGC** [comment]
@casigus @Kitakazi Can you download, extract and try this modified Zombie Cellar level I have made. This level I have forced to complete darkness (you can use the Flashlight to see).  All environmenals set to zero:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/f81e613d-5578-4d6c-bb31-243619fd4917)

Switched off auto exposure and other effects:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/1c41599a-d3dd-4f87-807f-e1b999b91e74)

Deleted ALL the lights from the level:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/1fdb12c6-c361-4738-93b0-7857551fcc2d)

When run, all is dark:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/c791f818-4e19-4ddd-aaa0-62fe675475af)

And flashlight to show we where are:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/9526a421-97a9-44ab-aedd-26e684557366)

Once you both confirm you can see what I see, we can then determine why your specific levels are unable to become completely black in the latest EXP build.

**Comment by Kitakazi** [comment]
I cannot speak for Casigus but this bug stopped happening to me a few weeks ago and I haven't had it happen since.

**Comment by LeeBamberTGC** [comment]
@Kitakazi Thanks for the feedback, I will mark as Fixed and wait for @casigus to respond.

**Comment by TenebreGaming** [comment]
I also have not had any issues with lighting recently, thanks to the great
work you are all doing!

**Comment by casigus** [comment]
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/68471279/bfe8580d-5c28-46a3-9842-339ddba78327)

yep this is still bugged for me on some levels. that should be pitch black and its not.

**Comment by LeeBamberTGC** [comment]
@casigus see previous post of mine to you.

**Comment by casigus** [comment]
ohhhhhh! ok wheres the modified map, wheres the download for it :)

**Comment by LeeBamberTGC** [comment]
@casigus Goo point. Zip was too big for GitHub, so find a link to the dowload. Let me know when you have it so I can delete, thanks: https://drive.google.com/file/d/1Ssus73MflwEqQpeXkND1F8153998ern8/view?usp=sharing

**Comment by casigus** [comment]
Got it. Ill get it loaded in and looked at!

**Comment by Kitakazi** [comment]
@LeeBamberTGC Hey this bug has started happening again after the latest EXP update. Like above, using the zombie cellar demo I removed all the lights, set the lighting color and ambient color to black. And it looks like this

![image](https://github.com/TheGameCreators/GameGuruRepo/assets/51464873/67623e8e-e52b-4a11-bec9-98f282b36a1c)
